packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -522,6 +715,7 @@ open class FileDownloader(context: Context, private val client: OkHttpClient) {
         .url(assetEntity.url!!.toString())
         .addHeadersFromJSONObject(assetEntity.extraRequestHeaders)
         .header("Expo-Platform", "android")
+        .header("Expo-Protocol-Version", "1")
```

Comment:
This is a breaking change for SDK 46, 47, and 48 (the next time we cut) since they always sent nothing (default 0). 

This means that they will now send v1 requests and get back directives and not have the JS API to represent directives (since that was a breaking change added in this PR, `isRollBackToEmbedded` for example).

Reply:
Background: 

We made a breaking API change to the expo-updates protocol and need to make the expo-updates library support that change (this PR). In doing so, it needs to communicate to the server that it speaks the new version dialect. 

But the new version dialect introduces new features that aren't supported by previous versions of the JS module (can't do a 1:1 backport).

So we're left with a few options:
1. Fork all the code per-sdk version (like iOS) to make those SDK versions speak 0 and only the new one speak 1. This works but makes working in the codebase a nightmare like working in iOS is.
2. Make this library speak both 0 and 1 and return results appropriately. I'm not sure this is possible. Then we could send a different accept version header depending on SDK version. This seems difficult but probably the most correct.

Reply:
Going to revert this and the rollbacks PR for now while we figure this out. In its current state, having this PR checked in is too risky.